src/components/UnitCard.js

```diff
@@ -12,27 +16,36 @@ const UpgradeBar = styled.div`
   display: flex;
 `;
 
-const RemoveButton = styled.button`
+const RemoveButton = styled.div`
   float: right;
 `;
 
-const UnitCard = ({ unit, deleteFunction, index }) => {
+const UnitCard = ({ unit, deleteFunction, index, updateFunction }) => {
   const { name, upgrades_slots } = unit;
   return (
     <Body>
       <p>
         {name}
-        <RemoveButton
-          onClick={() => {
-            deleteFunction(index);
-          }}
-        >
-          X{" "}
+        <RemoveButton>
+          <IconButton
+            onClick={() => {
+              deleteFunction(index);
+            }}
+          >
+            <ClearIcon />
+          </IconButton>
         </RemoveButton>
       </p>
       <UpgradeBar>
-        {upgrades_slots.map(name => (
-          <UpgradeButton type={name} />
+        {upgrades_slots.map((name, key) => (
+          <UpgradeButton
+            unitKey={index}
+            upgradeKey={key}
+            key
+            upgradeFunction={updateFunction}
+            type={name}
```

Comment:
Type is name? Might be nice to not use these interchangeably if they're actually the same thing